core/src/main/java/com/valtech/aem/saas/core/indexing/PageIndexUpdateHandler.java

```diff
@@ -0,0 +1,186 @@
+package com.valtech.aem.saas.core.indexing;
+
+import com.day.cq.commons.Externalizer;
+import com.day.cq.replication.ReplicationAction;
+import com.day.cq.replication.ReplicationActionType;
+import com.day.cq.replication.ReplicationEvent;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.google.common.collect.ImmutableMap;
+import com.valtech.aem.saas.api.caconfig.SearchConfiguration;
+import com.valtech.aem.saas.core.indexing.PageIndexUpdateHandler.Configuration;
+import com.valtech.aem.saas.core.resource.ResourceResolverProvider;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.caconfig.ConfigurationResolver;
+import org.apache.sling.event.jobs.Job;
+import org.apache.sling.event.jobs.JobManager;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+
+@Component(name = "Search as a Service - Page Replication Event Handler",
+    immediate = true,
+    service = {EventHandler.class},
+    configurationPolicy = ConfigurationPolicy.REQUIRE,
+    configurationPid = "com.valtech.aem.saas.core.indexing.IndexUpdateHandler",
+    property = {
+        EventConstants.EVENT_TOPIC + "=" + ReplicationAction.EVENT_TOPIC,
+        EventConstants.EVENT_TOPIC + "=" + ReplicationEvent.EVENT_TOPIC
+    })
+@Designate(ocd = Configuration.class)
+@Slf4j
+public class PageIndexUpdateHandler implements EventHandler {
+
+  public static final String SAAS_CONTENT_READER = "saas-content-reader";
```

Comment:
please inline constant in resourceResolverProvider. 
[] rename value to `saas-aem-module-service-reader`